common/main/cpp/native_c4replicator.cc

```diff
@@ -166,49 +166,50 @@ static bool pullFilterFunction(C4CollectionSpec, C4String, C4String, C4RevisionF
 
 static bool pushFilterFunction(C4CollectionSpec, C4String, C4String, C4RevisionFlags, FLDict, void *);
 
-// ??? Move the field stuff into initialization.
-static bool fromJavaReplColls(
+// ??? Move the field introspection into initialization.
+static int fromJavaReplColls(
         JNIEnv *env,
-        int nColls,
         jobjectArray jColls,
-        C4ReplicationCollection colls[],
+        std::vector<C4ReplicationCollection> colls,
```

Comment:
Without passing this by reference I fear this does not do what you think it does.  Passing by value actually makes a copy of the entire list, so you are just adding to a local one here, while leaving the original one unaffected.  I think what you want is to add it to the one you passed in right?  In that case use `std::vector<C4ReplicationCollection>& colls`